IA-1697: refactor InstancePopUpComponent as function

quang-le opened the pull request:
Old class component that need to be refactored in order to allow upgrading leaflet

Related JIRA tickets : IA-1697

## Self proof reading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests

## Changes

- replaced HOC with hooks for
    - redux state selectors
    - styling
    - translations
- wrapped callbacks in `useCallback`
- wrapped values computed in `render` with `useMemo` when necessary
- Fixed a styling bug with the "See" button, where only the link was actually clickable
- Use the new component in `InstancesMap` and `OrgUnitMapComponent`

## How to test

Log in, go to Instances > map tab, click on an instance to see the pop up. Click the "See" button: you should be redirected even if you don't click on the text

Go to Org units and do the same

Apart from the link, nothing should have changed

## Print screen / video

https://user-images.githubusercontent.com/38907762/214267904-f8152e00-dfac-42bf-892d-5badbb0490cf.mov

## Notes
merges into https://github.com/BLSQ/iaso/pull/328

beygorghor reviewed (approved):
Maybe just repove the `// eslint-disable-next-line no-unused-vars` line 184 on AddTaskComponent, i don't think it's used anymore

quang-le committed:
IA-1697: convert InstancesPopUp to TS

quang-le committed:
IA-1697: fix pop up button/link styling

quang-le committed:
IA-1697: use refactored component in Maps

- InstancesMap
- OrgUnitMapComponent

quang-le committed:
IA-1697: remove js file

quang-le committed:
Merge branch 'main' into IA-1696_OrgUnitMap_to_function

quang-le committed:
IA-1696: move const and helpers in utils

quang-le committed:
Ia-1696: add utils.ts

quang-le committed:
IA-1696: move SOurceShape in own component

quang-le committed:
Ia-1696: add OrgUnitMap.tsx

quang-le committed:
IA-1696: fix catchment group

quang-le committed:
Merge branch 'main' into IA-1696_OrgUnitMap_to_function

quang-le committed:
IA-1696: remove console error

quang-le committed:
IA-1696: fixlocation and catchment

quang-le committed:
IA-1696: remove unnecessary condition

quang-le committed:
IA-1696: add typing to mapUtils

quang-le committed:
IA-1696: add typing to utils

quang-le committed:
IA-1696: update outdated import

quang-le committed:
IA-1696: add constants.ts

quang-le committed:
IA-1696: refactor getMarkersList as MarkersList

quang-le committed:
IA-1696: add new custom types

quang-le committed:
IA-1696: move redux code in dedicated hook

quang-le committed:
IA-1696: split map elements in separate components

quang-le committed:
IA-1696: light cleanup

quang-le committed:
IA-1696: delete js component

quang-le committed:
Merge remote-tracking branch 'origin' into IA-1696_OrgUnitMap_to_function

quang-le committed:
Merge remote-tracking branch 'origin' into IA-1697_InstancesPopUp_to_function

quang-le committed:
IA-1696: add typing to MarkersList

quang-le committed:
IA-1696: reduce duplicated code

- replace SourcesSelectedMarkers, OrgUnitTypesSelectedMarkers
- with new SelectedMarkers component

quang-le committed:
Ia-1696: move buttonsInitialState to constants

quang-le committed:
IA-1696: fix PopUp button

quang-le committed:
IA-1686: add explanation comment

quang-le committed:
Merge branch 'IA-1697_InstancesPopUp_to_function' into IA-1696_OrgUnitMap_to_function

beygorghor commented:
@cypress

quang-le committed:
Merge remote-tracking branch 'origin' into IA-1697_InstancesPopUp_to_function

quang-le committed:
IA-1697: cleanup commented code

quang-le committed:
IA_1696: fix duplicate forms in filter dropdown

beygorghor committed:
Merge pull request #339 from BLSQ/IA-1696_OrgUnitMap_to_function

Ia 1696 org unit map to function
